### DISCUSSION TITLE
palette color union

### DISCUSSION THREAD

**Discussion by freeman1995** [chakra-ui/chakra-ui#6242] [Ideas]
what do u think of that:
![Screen Shot 2022-06-25 at 18 16 23](https://user-images.githubusercontent.com/16955580/175779831-93180993-3de4-4204-ab47-56345ef578b7.png)
<img width="1169" alt="Screen Shot 2022-06-25 at 14 25 30" src="https://user-images.githubusercontent.com/16955580/175779837-6798bec6-52aa-4751-b147-02e645a8141b.png">
<img width="826" alt="Screen Shot 2022-06-25 at 9 25 14" src="https://user-images.githubusercontent.com/16955580/175779839-e02717ae-5289-46c5-8da2-73f27583744a.png">

**Comment by Yair-Ohana** [comment]
This is amazing.

**Comment by freeman1995** [comment]
better way:
```
export type Join<K, P> = K extends string | number
  ? P extends string | number
    ? `${K}${"" extends P ? "" : "."}${P}`
    : never
  : never;

export type Leaves<T> = T extends object
  ? {
      [K in keyof T]: Join<K, Leaves<T[K]>>;
    }[keyof T]
  : "";
```
